framework/src/linearfvbcs/LinearFVBoundaryCondition.C

```diff
@@ -0,0 +1,106 @@
+#include "SystemBase.h"
+#include "MooseVariableFV.h"
+
+InputParameters
+LinearFVBoundaryCondition::validParams()
+{
+  InputParameters params = MooseObject::validParams();
+  params += TransientInterface::validParams();
+  params += BoundaryRestrictableRequired::validParams();
+  params += TaggingInterface::validParams();
+  params += NonADFunctorInterface::validParams();
+
+  MultiMooseEnum vtags("rhs time", "rhs", true);
+  auto & vector_tag_enum = params.set<MultiMooseEnum>("vector_tags", true);
+  vector_tag_enum = vtags;
+
+  params.addRequiredParam<LinearVariableName>(
+      "variable", "The name of the variable that this boundary condition applies to");
+  params.declareControllable("enable");
+  params.registerBase("LinearFVBoundaryCondition");
+  params.registerSystemAttributeName("LinearFVBoundaryCondition");
+  return params;
+}
+
+LinearFVBoundaryCondition::LinearFVBoundaryCondition(const InputParameters & parameters)
+  : MooseObject(parameters),
+    BoundaryRestrictableRequired(this, false),
+    SetupInterface(this),
+    FunctionInterface(this),
+    DistributionInterface(this),
+    UserObjectInterface(this),
+    TransientInterface(this),
+    PostprocessorInterface(this),
+    VectorPostprocessorInterface(this),
+    GeometricSearchInterface(this),
+    MeshChangedInterface(parameters),
+    TaggingInterface(this),
+    MooseVariableInterface(this,
+                           false,
+                           "variable",
+                           Moose::VarKindType::VAR_SOLVER,
+                           Moose::VarFieldType::VAR_FIELD_STANDARD),
+    MooseVariableDependencyInterface(this),
+    NonADFunctorInterface(this),
+    FaceArgProducerInterface(),
+    _tid(parameters.get<THREAD_ID>("_tid")),
+    _subproblem(*getCheckedPointerParam<SubProblem *>("_subproblem")),
+    _mesh(_subproblem.mesh()),
+    _fv_problem(*getCheckedPointerParam<FVProblemBase *>("_fe_problem_base")),
+    _var(*mooseLinearVariableFV()),
+    _sys(_var.sys()),
+    _linear_system(libMesh::cast_ref<libMesh::LinearImplicitSystem &>(_sys.system()))
+{
+  addMooseVariableDependency(&_var);
+}
+
+bool
+LinearFVBoundaryCondition::hasFaceSide(const FaceInfo & fi, bool fi_elem_side) const
+{
+  const auto ft = fi.faceType(std::make_pair(_var.number(), _var.sys().number()));
+  if (fi_elem_side)
+    return ft == FaceInfo::VarFaceNeighbors::ELEM || ft == FaceInfo::VarFaceNeighbors::BOTH;
+  else
+    return ft == FaceInfo::VarFaceNeighbors::NEIGHBOR || ft == FaceInfo::VarFaceNeighbors::BOTH;
+}
+
+Moose::FaceArg
+LinearFVBoundaryCondition::singleSidedFaceArg(const FaceInfo * fi,
+                                              const Moose::FV::LimiterType limiter_type,
+                                              const bool correct_skewness) const
+{
+  mooseAssert(fi, "FaceInfo should not be null!");
+  return makeFace(*fi, limiter_type, true, correct_skewness);
+}
+
+Real
+LinearFVBoundaryCondition::computeCellToFaceDistance() const
```

Comment:
this is more the ProjectedDistance or OrthonormalDistance than the distance

Reply:
From wikipedia:
```
The distance between two objects in physical [space](https://en.wikipedia.org/wiki/Space) is the [length](https://en.wikipedia.org/wiki/Length) of a [straight line](https://en.wikipedia.org/wiki/Line_segment) between them, which is the shortest possible path.
```
In this sense the naming we have is correct.